File: Data Structures & Algorithms/merge-intervals/submission-4.java
class Solution {
    public int[][] merge(int[][] intervals) {
        int maxi = 0;
        for (int[]  interval : intervals) {
            maxi = Math.max(interval[0], maxi);
        }
        int[] mp = new int[maxi + 1];
        for (int[] interval : intervals) {
            mp[interval[0]] = Math.max(mp[interval[0]], interval[1] + 1);
        }
        int have = -1;
        int start = -1;
        ArrayList<int[]> res = new ArrayList<>();
        for (int i = 0; i <= maxi; ++i) {
            if (mp[i] != 0) {
                if (start == -1) {
                    start = i;
                }
                    have = Math.max(mp[i] - 1, have);

            }
            if (i == have) {
                res.add(new int[]{start, have});
                have = -1;
                start = -1;
            }
        }
        if (start != -1) res.add(new int[]{start, have});
        return res.toArray(new int[res.size()][2]);
    }
}

File: Data Structures & Algorithms/meeting-schedule/submission-4.cpp
/**
 * Definition of Interval:
 * class Interval {
 * public:
 *     int start, end;
 *     Interval(int start, int end) {
 *         this->start = start;
 *         this->end = end;
 *     }
 * }
 */

class Solution {
public:
    bool canAttendMeetings(vector<Interval>& intervals) {
        sort(intervals.begin(), intervals.end(), [](auto& a, auto& b) {return a.start < b.start;});
        int prevEnd{intervals[0].end};
        for (int i{1}; i < intervals.size(); ++i) {
            int start = intervals[i].start;
            int end = intervals[i].end;
            if (start >= prevEnd) {
                prevEnd = end;
            }
            else {
                return false;
            }
        }
        return true;
    }
};
